bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/XMLFactory.java

```diff
@@ -239,7 +231,7 @@
      * @return Concatenated XML sequence
      */
     public static XMLValue<?> concatenate(XMLValue<?> firstSeq, XMLValue<?> secondSeq) {
-        ArrayValue concatSeq = new ArrayValue();
+        ArrayValue concatSeq = new ArrayValue(new BArrayType(firstSeq.getType()), -1);
```

Comment:
In the earlier version, ArrayType was missing from the ArrayValue. I got an NPE down the line. Is this the best way to solve this? @SupunS

Reply:
+1. can use the constructor that takes only the array-type.

I think we need to remove these additional constructors. Will create an issue for that.